### PR TITLE
add support for indexing (usize, usize)

### PR DESCRIPTION
**Pull request by BurntNail**
Add an `Index` and `IndexMut` impl for `(usize, usize)`

### CODE REVIEW

**Comment by BurntNail** [comment]
FYI this fixes a bunch of clippy-related issues

File: src/lib.rs [hunk 1]
```diff
@@ -832,14 +842,14 @@ impl<T: fmt::Debug> fmt::Debug for Grid<T> {
                         write!(
                             f,
                             " {item:width$.precision$?}",
-                            width = width,
-                            precision = precision
+                            // width = width,
```
Comment (becheran) [inline src/lib.rs:845]: Why did you commented out this?

**Review by becheran — COMMENTED** [review]
Thx for the PR. Love the linter improvements! Just the one question. Otherwise looks good to me

File: src/lib.rs (hunk 1 above)
Comment (BurntNail) [reply src/lib.rs:845]: It isn't necessary - format can pull in identifiers from the surrounding scope.

It's a relatively new feature - if you want to keep an older MSRV, just uncomment it.

Comment (becheran) [reply src/lib.rs:845]: Ah cool 😎 did not know that. Fine for me if it works

**Pull request merged**